.github/workflows/static-checks-self-hosted.yaml

```diff
@@ -0,0 +1,130 @@
+on:
+  pull_request_target:
+    branches:
+      - 'main'
+      - 'stable-*'
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - labeled # a workflow runs only when the 'ok-to-test' label is added
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+name: Static checks
+jobs:
+  build-checks:
+    strategy:
```

Comment:
The `build-checks` workflow is almost a complete copy of  https://github.com/kata-containers/kata-containers/blob/main/.github/workflows/static-checks.yaml . Changes on static-checks will be carried out on both files from now onward, which isn't good.

How feasible it would be to convert https://github.com/kata-containers/kata-containers/blob/main/.github/workflows/static-checks.yaml into a callable workflow (`workflow_call`)? Then both static-checks and static-checks-self-hosted would call it passing the `instance` as an input.

Reply:
Sounds much better. I will update the PR soon. Thanks for the feedback. 😉